v03_pipeline/lib/tasks/write_new_variants_table.py

```diff
@@ -0,0 +1,222 @@
+import functools
+
+import hail as hl
+import luigi
+
+from v03_pipeline.lib.annotations.annotations_utils import get_annotation_dependencies
+from v03_pipeline.lib.misc.util import callset_project_pairs
+from v03_pipeline.lib.model import Env, ReferenceDatasetCollection
+from v03_pipeline.lib.paths import (
+    new_variants_table_path,
+    remapped_and_subsetted_callset_path,
+    variant_annotations_table_path,
+)
+from v03_pipeline.lib.tasks.base.base_variant_annotations_table import (
+    BaseVariantAnnotationsTableTask,
+)
+from v03_pipeline.lib.tasks.base.base_write_task import BaseWriteTask
+from v03_pipeline.lib.tasks.files import GCSorLocalTarget
+from v03_pipeline.lib.tasks.reference_data.update_variant_annotations_table_with_updated_reference_dataset import (
+    UpdateVariantAnnotationsTableWithUpdatedReferenceDataset,
+)
+from v03_pipeline.lib.tasks.update_lookup_table import (
+    UpdateLookupTableTask,
+)
+from v03_pipeline.lib.tasks.write_remapped_and_subsetted_callset import (
+    WriteRemappedAndSubsettedCallsetTask,
+)
+
+VARIANTS_PER_VEP_PARTITION = 1e3
+
+
+class WriteNewVariantsTable(BaseWriteTask):
+    callset_paths = luigi.ListParameter()
+    project_guids = luigi.ListParameter()
+    project_remap_paths = luigi.ListParameter()
+    project_pedigree_paths = luigi.ListParameter()
+    ignore_missing_samples_when_subsetting = luigi.BoolParameter(
+        default=False,
+        parsing=luigi.BoolParameter.EXPLICIT_PARSING,
+    )
+    ignore_missing_samples_when_remapping = luigi.BoolParameter(
+        default=False,
+        parsing=luigi.BoolParameter.EXPLICIT_PARSING,
+    )
+    validate = luigi.BoolParameter(
+        default=True,
+        parsing=luigi.BoolParameter.EXPLICIT_PARSING,
+    )
+    liftover_ref_path = luigi.OptionalParameter(
+        default='gs://hail-common/references/grch38_to_grch37.over.chain.gz',
+        description='Path to GRCh38 to GRCh37 coordinates file',
+    )
+
+    @property
+    def annotation_dependencies(self) -> dict[str, hl.Table]:
+        return get_annotation_dependencies(self.dataset_type, self.reference_genome)
+
+    def output(self) -> luigi.Target:
+        return GCSorLocalTarget(
+            new_variants_table_path(
+                self.reference_genome,
+                self.dataset_type,
+            ),
+        )
+
+    def requires(self) -> list[luigi.Task]:
+        upstream_table_tasks = [
+            BaseVariantAnnotationsTableTask(
+                self.reference_genome,
+                self.dataset_type,
+                self.sample_type,
+            ),
+        ]
+        if Env.REFERENCE_DATA_AUTO_UPDATE:
+            upstream_table_tasks.append(
+                UpdateVariantAnnotationsTableWithUpdatedReferenceDataset(
+                    self.reference_genome,
+                    self.dataset_type,
+                    self.sample_type,
+                ),
+            )
+        if self.dataset_type.has_lookup_table:
```

Comment:
This is true. I had both here so that `BaseVariantAnnotationsTableTask` would call `initialize_table` to create an empty annotations table in test but there's a better way to do that

Reply:
Does `UpdateVariantAnnotationsTableWithUpdatedReferenceDataset` call `initialize_table` under the hood?

Reply:
My previous answer was wrong - ` initialize_table` is called but also `update_table` is as well, and in `BaseVariantAnnotationsTableTask` `update_table` simply returns `ht`. If I don't mock out the `UpdateVariantAnnotationsTableWithUpdatedReferenceDataset` dependency in tests it runs its implementation of `update_table` which is a can of worms and doesn't need to happen in those tests 

I have an idea of how to fix it though